enzyme/Enzyme/MLIR/Implementations/LLVMAutoDiffOpInterfaceImpl.cpp

```diff
@@ -153,7 +185,11 @@ struct StoreOpInterfaceReverse
 
   SmallVector<Value> cacheValues(Operation *op,
                                  MGradientUtilsReverse *gutils) const {
-    return SmallVector<Value>();
+    auto storeOp = cast<LLVM::StoreOp>(op);
+    auto addr = storeOp.getAddr();
```

Comment:
if the pop is gated by `if (!gutils->isConstantValue(addr))`, the push also needs to have the same condition